presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -76,4 +77,13 @@ else if (newSize > MAX_ARRAY_SIZE) {
         }
         return (int) newSize;
     }
+
+    static int calculateBlockResetBytes(int currentBytes)
+    {
+        long newBytes = (long) ceil(currentBytes * BLOCK_RESET_SKEW);
+        if (newBytes > Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
```

Comment:
In Java there is JVM specific limit for arrays that is a few entries less than `Integer.MAX_VALUE`: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size

I'd use something like `-16` to be safe

Reply:
Presto is already exposed to this. We  have MAX_ARRAY_SIZE in BlockUtil. We used -8, which was copied from ArrayList.MAX_ARRAY_SIZE.